pyro/distributions/transforms/matrix_exponential.py

```diff
@@ -160,15 +163,157 @@ def reset_parameters(self):
+
+    Example usage:
+
+    >>> from pyro.nn.dense_nn import DenseNN
+    >>> input_dim = 10
+    >>> context_dim = 5
+    >>> batch_size = 3
+    >>> base_dist = dist.Normal(torch.zeros(input_dim), torch.ones(input_dim))
+    >>> param_dims = [input_dim*input_dim]
+    >>> hypernet = DenseNN(context_dim, [50, 50], param_dims)
+    >>> transform = ConditionalMatrixExponential(input_dim, hypernet)
+    >>> z = torch.rand(batch_size, context_dim)
+    >>> flow_dist = dist.ConditionalTransformedDistribution(base_dist,
+    ... [transform]).condition(z)
+    >>> flow_dist.sample(sample_shape=torch.Size([batch_size])) # doctest: +SKIP
+
+    :param input_dim: the dimension of the input (and output) variable.
+    :type input_dim: int
+    :param iterations: the number of terms to use in the truncated power series that
+        approximates matrix exponentiation.
+    :type iterations: int
+    :param normalization: One of `['none', 'weight', 'spectral']` normalization that
+        selects what type of normalization to apply to the weight matrix. `weight`
+        corresponds to weight normalization (Salimans and Kingma, 2016) and
+        `spectral` to spectral normalization (Miyato et al, 2018).
+    :type normalization: string
+    :param bound: a bound on either the weight or spectral norm, when either of
+        those two types of regularization are chosen by the `normalization`
+        argument. A lower value for this results in fewer required terms of the
+        truncated power series to closely approximate the exact value of the matrix
+        exponential.
+    :type bound: float
+
+    References:
+
+    [1] Emiel Hoogeboom, Victor Garcia Satorras, Jakub M. Tomczak, Max Welling. The
+        Convolution Exponential and Generalized Sylvester Flows. [arXiv:2006.01910]
+    [2] Tim Salimans, Diederik P. Kingma. Weight Normalization: A Simple
+        Reparameterization to Accelerate Training of Deep Neural Networks.
+        [arXiv:1602.07868]
+    [3] Takeru Miyato, Toshiki Kataoka, Masanori Koyama, Yuichi Yoshida. Spectral
+        Normalization for Generative Adversarial Networks. ICLR 2018.
+
+    """
+
+    domain = constraints.real_vector
+    codomain = constraints.real_vector
+    bijective = True
+    event_dim = 1
+
+    def __init__(self, input_dim, nn, iterations=8, normalization='none', bound=None):
+        super().__init__()
+        self.input_dim = input_dim
+        self.nn = nn
+        self.iterations = iterations
+        self.normalization = normalization
+        self.bound = bound
+
+    def _params(self, context):
+        return self.nn(context)
+
+    def condition(self, context):
+        # This hack could be fixed by having a conditioning network that outputs a more general shape
+        cond_nn = partial(self.nn, context)
+
+        def weights():
+            w = cond_nn()
+            return w.view(w.shape[:-1] + (self.input_dim, self.input_dim))
```

Comment:
FYI local functions are not serializable. I'm unsure whether these conditional transforms are jit scriptable.

Beyond the scope of this PR, we might consider adding some jit script tests to test_transfors.py...